src/screens/Game/TriviaMain.tsx

```diff
@@ -51,32 +74,52 @@ const styles = StyleSheet.create({
 const TOTAL_TIME = gameDescriptions.Trivia.minutes * 60;
 type Props = NativeStackScreenProps<RootStackParamList, "TriviaMain">;
 
-function TriviaScreen({ navigation, route }: Props) {
+export default function TriviaScreen({ navigation, route }: Props) {
   const [problem, setProblem] = useState(getProblem());
   const [answered, setAnswered] = useState(false);
+  const { updateStatsOnAnswer, onTimeComplete } = useTriviaProblems({
+    navigation,
+    route,
+  });
 
-  const getNewProblem = () => {
+  const elapsedTimeRef = useRef<number>(0);
+  const remainingTimeRef = useRef<RemainingTimeGetter>();
+
+  const getNewProblem = useCallback(() => {
     setProblem(getProblem());
     setAnswered(false);
-  };
+    elapsedTimeRef.current = 0;
+  }, []);
 
   const onPressShowButton = useCallback(() => {
     if (!answered) {
       setAnswered(true);
     } else {
       getNewProblem();
     }
-  }, [answered]);
+  }, [answered, getNewProblem]);
+
+  const onPressYesButton = useCallback(() => {
+    elapsedTimeRef.current =
+      remainingTimeRef.current.getRemainingTime() - TOTAL_TIME;
+    updateStatsOnAnswer(true, elapsedTimeRef.current);
+    getNewProblem();
+  }, [getNewProblem, updateStatsOnAnswer]);
 
-  const onTimeComplete = useCallback(() => {
-    navigation.navigate(...route.params.nextScreenArgs);
-  }, [navigation, route.params.nextScreenArgs]);
+  const onPressNoButton = useCallback(() => {
+    elapsedTimeRef.current =
+      remainingTimeRef.current.getRemainingTime() - TOTAL_TIME;
+    updateStatsOnAnswer(false, elapsedTimeRef.current);
+    setAnswered(true);
+    getNewProblem();
+  }, [getNewProblem, updateStatsOnAnswer]);
```

Comment:
Since these two functions perform the same function on a high level, merge them into one function called `onAnswerClick` which takes in a parameter `isCorrect: boolean`